error TS2532: Object is possibly 'undefined'

### Research

- [X] I searched but did not find an existing issue or discussion about this bug.

### Description

Tried it with npm V17. and V16.
Downloaded the example demos but can't get the examples running.


```
/etc/demofile/examples # npx ts-node dumpfile.ts ../demos/pc419-vs-chiefs-mirage.dem

/etc/demofile/node_modules/ts-node/src/index.ts:500
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
dumpfile.ts:61:9 - error TS2532: Object is possibly 'undefined'.

61         terrorists.teamName,
           ~~~~~~~~~~
dumpfile.ts:62:9 - error TS2532: Object is possibly 'undefined'.

62         terrorists.clanName,
           ~~~~~~~~~~
dumpfile.ts:63:9 - error TS2532: Object is possibly 'undefined'.

63         terrorists.score,
           ~~~~~~~~~~
dumpfile.ts:64:9 - error TS2532: Object is possibly 'undefined'.

64         cts.teamName,
           ~~~
dumpfile.ts:65:9 - error TS2532: Object is possibly 'undefined'.

65         cts.clanName,
           ~~~
dumpfile.ts:66:9 - error TS2532: Object is possibly 'undefined'.

66         cts.score
           ~~~
dumpfile.ts:191:15 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.

191 parseDemoFile(process.argv[2]);
                  ~~~~~~~~~~~~~~~

    at createTSError (/etc/demofile/node_modules/ts-node/src/index.ts:500:12)
    at reportTSError (/etc/demofile/node_modules/ts-node/src/index.ts:504:19)
    at getOutput (/etc/demofile/node_modules/ts-node/src/index.ts:739:36)
    at Object.compile (/etc/demofile/node_modules/ts-node/src/index.ts:955:32)
    at Module.m._compile (/etc/demofile/node_modules/ts-node/src/index.ts:1043:43)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/etc/demofile/node_modules/ts-node/src/index.ts:1046:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)

```

### Code to reproduce

_No response_

### Affected demos

_No response_

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -52,7 +52,11 @@ jobs:
         run: |
           npm uninstall demofile
           npm install ../demofile-*.tgz
-          npx ts-node scores.ts ../demos/pc419-vs-chiefs-mirage.dem
+          for file in *.ts
+          do
+            echo "Running example: $file..."
+            npx ts-node "$file" ../demos/pc419-vs-chiefs-mirage.dem
+          done
 
   lint:
     name: Linting

--- a/examples/dumpfile.ts
+++ b/examples/dumpfile.ts
@@ -53,8 +53,8 @@ function parseDemoFile(path: string) {
     function logTeamScores() {
       const teams = demoFile.teams;
 
-      const terrorists = teams[TeamNumber.Terrorists];
-      const cts = teams[TeamNumber.CounterTerrorists];
+      const terrorists = teams[TeamNumber.Terrorists]!;
+      const cts = teams[TeamNumber.CounterTerrorists]!;
 
       console.log(
         "\t%s: %s score %d\n\t%s: %s score %d",
@@ -188,4 +188,4 @@ function parseDemoFile(path: string) {
   });
 }
 
-parseDemoFile(process.argv[2]);
+parseDemoFile(process.argv[2]!);

--- a/examples/join-leave.ts
+++ b/examples/join-leave.ts
@@ -76,4 +76,4 @@ function parseDemoFile(path: string) {
   });
 }
 
-parseDemoFile(process.argv[2]);
+parseDemoFile(process.argv[2]!);

--- a/examples/molotov.ts
+++ b/examples/molotov.ts
@@ -63,4 +63,4 @@ function parseDemoFile(path: string) {
   });
 }
 
-parseDemoFile(process.argv[2]);
+parseDemoFile(process.argv[2]!);

--- a/examples/package.json
+++ b/examples/package.json
@@ -7,6 +7,6 @@
   "license": "ISC",
   "dependencies": {
     "demofile": "file:..",
-    "typescript": "^4.1.3"
+    "typescript": "^4.4.3"
   }
 }

--- a/examples/package.json
+++ b/examples/package.json
@@ -7,6 +7,6 @@
   "license": "ISC",
   "dependencies": {
     "demofile": "file:..",
-    "typescript": "^4.4.3"
+    "typescript": "~4.4.3"
   }
 }

--- a/examples/plant-site.ts
+++ b/examples/plant-site.ts
@@ -45,4 +45,4 @@ function parseDemoFile(path: string) {
   });
 }
 
-parseDemoFile(process.argv[2]);
+parseDemoFile(process.argv[2]!);

--- a/examples/purchases.ts
+++ b/examples/purchases.ts
@@ -52,4 +52,4 @@ function parseDemoFile(path: string) {
   });
 }
 
-parseDemoFile(process.argv[2]);
+parseDemoFile(process.argv[2]!);

--- a/examples/rank.ts
+++ b/examples/rank.ts
@@ -51,4 +51,4 @@ function parseDemoFile(path: string) {
   });
 }
 
-parseDemoFile(process.argv[2]);
+parseDemoFile(process.argv[2]!);

--- a/examples/tickrate.ts
+++ b/examples/tickrate.ts
@@ -35,4 +35,4 @@ function parseDemoFile(path: string) {
   });
 }
 
-parseDemoFile(process.argv[2]);
+parseDemoFile(process.argv[2]!);

--- a/examples/tsconfig.json
+++ b/examples/tsconfig.json
@@ -1,0 +1,7 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "."
+  },
+  "include": ["*.ts"]
+}

--- a/package.json
+++ b/package.json
@@ -76,7 +76,7 @@
     "ts-node": "^9.0.0",
     "ts-proto": "^1.41.1",
     "typedoc": "^0.19.2",
-    "typescript": "^4.4.3"
+    "typescript": "~4.4.3"
   },
   "husky": {
     "hooks": {

